# Assignment 01
##### Python Basics I

This tutorial was written by Terry Ruas (University of Göttingen). The references for external contributors for which this material was anyhow adapted/inspired are in the Acknowledgments section (end of the document).


## Task 01 – Hello World!
Write a simple program that asks the user to input a name and outputs a simple greeting, such as ‘Hi Name! Nice to meet you!’ and in the other line, ‘Welcome to the Programing Course!’. 

*Hint: Check out the input () and print() function in Python.* 

In [3]:
b = input("Hello, what is your name?\n")                                        #Prints the "txt" and gets a user input
print("Hi, " + b + "! Nice to meet you!\nWellcome to the Programing Course!")   #Prints the given input with a static text around it

Hi, Lukas! Nice to meet you!
Wellcome to the Programing Course!


## Task 02 – Reversed Words
Write a program that reads in a word provided by the user and prints the word in a reversed order. For example, if the input is hello, the output should be ‘olleh’. 

- Use a loop structure 
- Use just array indexes 


In [4]:
a = input("You type a Word and type it backwords :)\n")         #Get a input from the user in cmd
print (a[::-1])                                                 #prints it backwards

ztas nie reih ollaH


## Task 03 –  Fibonacci Numbers
Write a program that reads in an upper bound (number) provided by the user and prints the sequence of Fibonacci numbers that are less or equal to the number given by the user. Use a while-loop for this task. 

The Fibonacci sequence is defined as 
$ F_{n}=F_{n−1}+F_{n−2}$. Use $F_{0}=0$ and $F_{1} = 1$ as seed values.  

For example, if the user inputs 6, the output should be 0, 1, 1, 2, 3, 5. 





In [7]:
f0 = 0
f1 = 1
fn = 1

number = int(input())
string = []

while f0 < number:
    string.append(f0)
    # print(f0)
    fn = f0 + f1
    f0 = f1
    f1 = fn

print(*string)


0 1 1 2 3 5 8 13 21 34 55 89 144 233 377 610


## Task 04 – Selective Printing 

Write a program that reads in an upper bound (number) provided by the user and prints all integer numbers that are less or equal to the upper bound except the integer numbers that are divisible by 3. Use the continue statement. For example, if the user enters 6 as the upper bound, the output should be 0, 1, 2, 4, 5.  

In [9]:
number = int(input())

i = 0
print(i)

while i <= number:

    if (i % 3 != 0):
        print(i)
    i += 1


0
1
2
4
5
7
8
10
11
13
14
16
17
19
20
22
23
25
26
28
29
31
32
34
35
37
38
40
41
43
44
46
47
49
50
52
53
55
56
58
59
61
62
64
65
67
68
70
71
73
74
76
77
79
80
82
83
85
86
88
89
91
92
94
95
97
98
100
101
103
104
106
107
109
110
112
113
115
116
118
119
121
122
124
125
127
128
130
131
133
134
136
137
139
140
142
143
145
146
148
149
151
152
154
155
157
158
160
161
163
164
166
167
169
170
172
173
175
176
178
179
181
182
184
185
187
188
190
191
193
194
196
197
199
200
202
203
205
206
208
209
211
212
214
215
217
218
220
221
223
224
226
227
229
230
232
233
235
236
238
239
241
242
244
245
247
248
250
251
253
254
256
257
259
260
262
263
265
266
268
269
271
272
274
275
277
278
280
281
283
284
286
287
289
290
292
293
295
296
298
299
301
302
304
305
307
308
310
311
313
314
316
317
319
320
322
323
325
326
328
329
331
332
334
335
337
338
340
341
343
344
346
347
349
350
352
353
355
356
358
359
361
362
364
365
367
368
370
371
373
374
376
377
379
380
382
383
385
386
388
389
391
392
394
395
397
398
400
40

## Task 05 – Triangle Checking   

Write a Python program that asks the user to input the lengths of the sides in a triangle and outputs whether the triangle is equilateral, isosceles, or scalene. The program should also check for the Triangle inequality $(z < x + y)$,  and prompt the user for a valid input. 

- An equilateral triangle is a triangle in which all three sides are of equal length. 
- An isosceles triangle is a triangle with (at least) two sides of equal length. 
- A scalene triangle is a triangle in which all three sides have different lengths. 

For example, if the user inputs a=3, b=4, c=5, the triangle is scalene. 


In [11]:
a = int(input("input number a\n"))
b = int(input("input number b\n"))
c = int(input("input number c\n"))

if ((a <= b + c) and (b <= a + c) and (c <= b + a)):
    print("triangle inequality valid")
    if (a == b == c):
        print("equilateral")

    elif ((a == b) or (a == c) or (b == c)):
        print("isoscleces")

    else:
        print("scalene")
else:
    print("INVALID INPUT, try again")


triangle inequality valid
scalene


## Task 06 – Decimal to Octal Conversion 

Write a program that reads a non-negative integer number in the decimal representation provided by the user and prints the octal representation of the number. For example, if the user enters 167, the output should be 247.  

Write a more general program that reads in a non-negative number (potentially including decimal places) in the decimal representation provided by the user and prints the octal representation of the number. For example, if the user enters 25.11, the output should be 31.0702436560507534. 

In [ ]:
int = int(input())
# x = oct(int)
str = []


# print(x[2:])

i = 0
j = int

while 8**i <= int:
    str.append(j % 8)
    # print(j%8)
    j //= 8
    i += 1

print(*str[::-1])

# a

float = float(input())
str2 = []
k = 0
l = float

while 8**k <= float:
    str2.append(l % 8)
    # print(j%8)
    l /= 8
    k += 1

print(*str2[::-1])


## Acknowledgements

Redmond, Hsu, Saini, Gupta, Ramsey, Kondrich, Capoor, Cohen, Borus, Kincaid, Malik, and many others. - Stanford CS41